include/ifc/pathname.hxx

```diff
@@ -103,8 +104,9 @@ namespace ifc {
 
         value_type back() const
         {
-            IFCASSERT(length() > 0);
-            return begin()[length() - 1];
+            std::size_t index = length();
+            IFCASSERT(index != 0);
```

Comment:
`index` is not really the index into the sequence.  Use `auto` to preserve whatever the return type is.  I would just write that as
```
const auto n = length();
IFASSERT(n > 0);
return begin()[n - 1];
```